### PR TITLE
deps: upgrade sdk to v0.35.0

### PR DESCRIPTION
**Pull request by rflechtner**
It's time for an sdk upgrade!

Fully compliant DID resolution is now a feature of the sdk; the did driver thus only needs to provide http bindings and produce json-ld or json representations of the document depending on the requested MIME type.

~NOTE: due to changes introduced with sdk v0.29, this PR changes how keys are referenced by verification relationship entries. Ids are now relative (starting with a #).~
^ In order not to break expectations and any applications that may depend on the current behaviour, I changed this.

### CODE REVIEW

**Review by ntn-x2 — COMMENTED** [review]
Overlook it is a nice cleaning! Please before merging, remember to pull in the changes included in https://github.com/KILTprotocol/sdk-js/pull/725.

Also, when opening a PR against the uniresolver repo, please update the example light DIDs to something else more.... correct 😄😄😄

**Comment by rflechtner** [comment]
Good point, let's wait for the next patch release then.

**Review by ntn-x2 — APPROVED** [review]

**Comment by rflechtner** [comment]
I think we're changing how key ids are represented on the DID doc, so we should maybe wait until that is released?

**Comment by rflechtner** [comment]
@ntn-x2 the behaviour of the sdk has not changed, so I patched it on this end - ids are expanded by the resolver.

File: src/index.js
Comment (Dudleyneedham) [inline src/index.js:45]: IS this a new function??

File: src/index.js
Comment (Dudleyneedham) [inline src/index.js:101]: What is this mime type?

**Review by Dudleyneedham — APPROVED** [review]
LGTM but some minor questions

File: src/index.js
Comment (ntn-x2) [inline src/index.js:58]: Mhh, why did we need the key IDs to be expanded? Was it causing some issues with JSON-LD or VC proof verification?

**Review by ntn-x2 — APPROVED** [review]
Very much needed refresh, althought it won't last long before the release of the SDK v1! P.S. Now the did `did:kilt:light:004pqDzaWi3w7TzYzGnQDyrasK6UnyNnW6JQvWRrq6r8HzNNGy:z14mMLbhZGB6YYU7ud2eFvUiHz3Mwo6UdttffCxB5s4hB3pxV2UgTQrgTyV6MZ8FAvqqKZQpxsJTFRYHzYhjzDUbxMtyxQtTrBu4F9YZx99AuEHuNSPCCd8RqpLeczkuDTGMP7eBDmmNbPbiXhKv5hb6ibYPCpZjUtjPBDqUQ1wXmBv3` does not fail on the old resolver deployed in the uniresolver while it fails here, so we might want to change the example one to something else, that we can take from the SDK tests, as long as it has an encryption key and one service.

**Comment by ntn-x2** [comment]
Also, out of scope for this PR, but we could make tickets to: migrate from JS to TS and, write basic unit tests to make sure that the example URIs in the resolver work with the current version of the driver. What do you think?

Also, on a nice note, results on the uniresolver can now be checked for compliance, and ours turns lights green ✅✅ https://didlint.ownyourdata.eu/validate?did=did:kilt:4rNTX3ihuxyWkB7wG3oLgUWSBLa2gva1NBKJsBFm7jJZUYfc

File: src/index.js
Comment (rflechtner) [reply src/index.js:58]: Kinda both. The overwhelming majority of third-party software cannot handle this, and this includes json-ld processors if no 'base' is set.

**Comment by rflechtner** [comment]
> Also, out of scope for this PR, but we could make tickets to: migrate from JS to TS and, write basic unit tests to make sure that the example URIs in the resolver work with the current version of the driver. What do you think?

That does make sense!

File: src/index.js
Comment (rflechtner) [reply src/index.js:45]: it's 'new' in the sdk 0.31.0 or so. we just never updated the did driver

File: src/index.js
Comment (rflechtner) [reply src/index.js:101]: the mime type for the structure of did document and metadata called 'did resolution result'. It's defined in the did resolution specs

**Pull request merged**
